### PR TITLE
Add no_return_type to DemangleOptions.

### LINKED ISSUE #201: Ability to hide the return type?
Hi,

I'm ending up with demangled names such as these ones:

 - `void draw_quad_spans<unsigned int>(int, glsl::vec2_scalar*, unsigned short, float __vector(16)*, Texture&, int, Texture&, ClipRect const&)` (from mangled name: `_ZL15draw_quad_spansIjEviPN4glsl11vec2_scalarEtPDv16_fR7TextureiS6_RK8ClipRect`)
 - `void draw_elements<unsigned short>(int, int, Buffer&, unsigned long, VertexArray&, Texture&, int, Texture&)` (from mangled name: `_ZL13draw_elementsItEviiR6BuffermR11VertexArrayR7TextureiS5_`)
 - `int check_depth8<515, false>(unsigned short, unsigned short*, short __vector(8)&)` (from mangled name: `_ZL12check_depth8ILi515ELb0EEitPtRDv8_s`)
 - `decltype ((applyImpl)({parm#1}, {parm#2}, (*this).mArguments, std::__1::integer_sequence<unsigned long, (unsigned long)0, (unsigned long)1>{})) mozilla::detail::RunnableMethodArguments<mozilla::wr::WrWindowId, bool>::apply<mozilla::wr::RenderThread, void (mozilla::wr::RenderThread::*)(mozilla::wr::WrWindowId, bool)>(mozilla::wr::RenderThread*, void (mozilla::wr::RenderThread::*)(mozilla::wr::WrWindowId, bool))` (from mangled name: `_ZN7mozilla6detail23RunnableMethodArgumentsIJNS_2wr10WrWindowIdEbEE5applyINS2_12RenderThreadEMS6_FvS3_bEEEDTcl9applyImplfp_fp0_dtdefpT10mArgumentstlNSt3__116integer_sequenceImJLm0ELm1EEEEEEEPT_T0_`)

I would prefer to hide the return type. Especially in the last case it produces a lot of visual noise. Also, most symbols do not have a return type, so skipping return types for all symbols would make things more consistent and uniform.

What are your opinions on adding a flag to skip the return type?

### PR DESCRIPTION
**Pull request by mstange**
Fixes #201.

I could use some feedback on the naming; I flipped back and forth between `no_return`, `no_return_type` and `no_return_types` and then settled on `no_return_type`.

### CODE REVIEW

File: c_api/include/cpp_demangle.h [hunk 1]
```diff
@@ -8,6 +8,8 @@ extern "C" {
 struct DemangleOptions {
   // Do not display function arguments.
   bool no_params;
+  // Do not display the function return type.
+  bool no_return_type;
```
Comment (khuey) [inline c_api/include/cpp_demangle.h:12]: I think the naming is fine. A single function has multiple parameters but only one return type, so this makes sense.

**Comment by mstange** [comment]
Thanks!

**Comment by mstange** [comment]
@khuey Could you release a version that contains this patch?

**Comment by khuey** [comment]
I'll take a look at it tonight.

**Comment by khuey** [comment]
0.2.17 released.

**Comment by mstange** [comment]
Thank you very much!

**Pull request merged**